proto/v1/database_service.proto

```diff
@@ -1319,6 +1319,9 @@ message ChangedResourceSchema {
 
 message ChangedResourceTable {
   string name = 1;
+
+  // The ranges of sub-strings correspond to the statements on the sheet.
+  repeated Range ranges = 3;
```

Comment:
Comment 0-based or 1-based?

Reply:
The ranges are 0 based indices by default so no confusing here.